test/e2e/validation_test.go

```diff
@@ -815,6 +815,38 @@ func (tc *testContext) testNodeAnnotations(t *testing.T) {
 	}
 }
 
+// testNodeCloudProviderFields tests that all required labels and fields are on each Windows node managed by the cloud
+// provider controller
```

Comment:
What is the cloud provider controller?

Reply:
WRT to BYOH nodes, if they are running on the same infrastructure as the rest of the cluster, as they need to be if the cluster's platform != none, they will be integrated with the cloud provider.

Reply:
> What is the cloud provider controller?

Re-worded

Reply:
> WRT to BYOH nodes, if they are running on the same infrastructure as the rest of the cluster, as they need to be if the cluster's platform != none, they will be integrated with the cloud provider.

@sebsoto  I double-checked the "BYOH"  nodes we are provisioning as part of the e2e test contains the providerID.

Considering those are not true "BYOH" as machines back them, the check will pass.

Are you proposing to check all nodes, or keep it to only Windows nodes backed by machine controller.

Reply:
@jrvaldes It is my understanding that CCM has no concept of an OpenShift 'Machine'. As long as it is a VM on the cloud provider (which it always should be in our CI), we should be running the checks.

Reply:
addressed

https://github.com/openshift/windows-machine-config-operator/compare/481eed0693140d61fd1ed5ad47d5784425204895..c7269495ac4fd5fcec24bba99c4723dcc8d33635